### PR TITLE
W-13749792: Fix issue validating objects array at multipart

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: src/test/java/org/mule/module/apikit/validation/body/form/AbstractMultipartRequestValidatorTestCase.java
Comment (dcourtil) [inline src/test/java/org/mule/module/apikit/validation/body/form/AbstractMultipartRequestValidatorTestCase.java:1]: ❤️

File: src/test/resources/munit/body/form/multipart-form-array/library/sample-types.raml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          attribute: true
+        type: string
+      value:
+        xml:
+          attribute: true
+        type: string
+
+  Metadata:
+    displayName: Sample MetaData
+    xml:
```
Comment (dcourtil) [inline src/test/resources/munit/body/form/multipart-form-array/library/sample-types.raml:25]: I see the example as a JSON, Is this intended to be an xml? Wondering if the xml node makes a difference for the validation

Comment (juanchib) [reply src/test/resources/munit/body/form/multipart-form-array/library/sample-types.raml:25]: Yes you're right I need to remove that an other this at the api/test that are related to the customer. Will do.

**Review by iglosiggio — DISMISSED** [review]

File: src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormDataTextParameter.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public MultipartFormDataTextParameter(String body) {
   @Override
   public void validate(Parameter expected) throws InvalidFormParameterException {
     if (expected.isArray()) {
-      if (!expected.validateArray(Collections.singletonList(body))) {
+      if (!expected.validateArray(Collections.singletonList(body)) && !expected.validate(body)) {
         throw new InvalidFormParameterException(expected.message(body));
       }
     } else if (!expected.validate(body)) {
```
Comment (iglosiggio) [inline src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormDataTextParameter.java:32]: What if we replace all this with:
```java
    boolean isArray = expected.isArray();
    boolean isInvalidArray = isArray && !expected.validateArray(Collections.singletonList(body)) && !expected.validate(body);
    boolean isInvalidNonArray = !isArray && !expected.validate(body);
    if (isInvalidArray || isInvalidNonArray) {
      throw new InvalidFormParameterException(expected.message(body));
    }
```

Comment (juanchib) [reply src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormDataTextParameter.java:32]: I like it it,, less spaghetti

**Review by iglosiggio — APPROVED** [review]

**Review by iglosiggio — APPROVED** [review]

File: src/test/java/org/mule/module/apikit/validation/body/UnionXMLBodyTestCase.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
 import org.mule.module.apikit.api.exception.BadRequestException;
 import org.mule.module.apikit.api.exception.MuleRestException;
 import org.mule.module.apikit.validation.AbstractRequestValidatorTestCase;
+import org.mule.module.apikit.validation.TestRestRequestValidatorBuilder;
```
[the remark this answers is not in the document]
Comment (juanchib) [reply src/test/java/org/mule/module/apikit/validation/body/UnionXMLBodyTestCase.java:15]: no prob! I did another run and I found a bunch of other un used imports! thx!

**Review by dcourtil — APPROVED** [review]

**Pull request merged**